src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/DotNetCoreProjectCompatibilityDetector.cs

```diff
@@ -335,35 +335,61 @@ private static CompatibilityLevel GetCompatibilityLevelFromVersion(Version versi
                 version = new Version(version.Major, version.Minor);
             }
 
-            if (compatData.SupportedVersion != null)
+            if (compatData.SupportedPreviewVersion is null &&
+                compatData.SupportedVersion is null &&
+                compatData.UnsupportedVersion is null)
             {
-                if (version < compatData.SupportedVersion)
-                {
-                    return CompatibilityLevel.Recommended;
-                }
-                else if (version == compatData.SupportedVersion || (compatData.UnsupportedVersion != null && version < compatData.UnsupportedVersion))
-                {
-                    return CompatibilityLevel.Supported;
-                }
+                // no restrictions
```

Comment:
Nit: throught - normally we capitalize the first letter of comments (applies throughout)